test/integration/targets/file/tasks/main.yml

```diff
@@ -258,6 +258,17 @@
 - name: copy directory structure over
   copy: src=foobar dest={{output_dir}}
 
+- name: check what would be removed if folder state was absent and diff is enabled
+  file: path={{output_dir}} state=absent
+  check_mode: yes
+  diff: yes
+  register: folder_absent_result
+
+- name: assert that the absent check lists expected files and folders
+  assert:
+    that:
+      - folder_absent_result.diff.before.path_content is defined
```

Comment:
I would also loosely verify the contents of the variable such as checking the length of the list or looking for a string in the list contents.

Reply:
Thanks, I'll figure out an something appropriate test and update when I get a chance

Reply:
Hey @samdoran,  
I've added a couple of additional asserts to validate that a known folder and file are included in the output. I believe this resolves the conversation, but if not please do re-open it